### PR TITLE
Integrate TransportURL when deploying Manila

### PR DESCRIPTION
**Pull request by fmount**
`TransportURL` CR provides a `transport_url` value needed to configure Manila (otherwise manila-share won't start).
`TransportURL` secret is provided by openstack-operator which deploys RabbitMQ and wait for it to be ready: at that point a TransportURL secret is created and can be consumed by Manila.

### CODE REVIEW

**Comment by fmount** [comment]
Ok, @gouthampacha I think we should be ready to go here!

File: controllers/manila_controller.go [hunk 1]
```diff
@@ -180,15 +186,63 @@ func (r *ManilaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ManilaReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// transportURLSecretFn - Watch for changes made to the secret associated with the RabbitMQ
```
Comment (gouthampacha) [inline controllers/manila_controller.go:189]: Thank you for adding this explanation

File: controllers/manila_controller.go [hunk 2]
```diff
@@ -180,15 +186,63 @@ func (r *ManilaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
...
 		For(&manilav1beta1.Manila{}).
 		Owns(&mariadbv1.MariaDBDatabase{}).
 		Owns(&manilav1beta1.ManilaAPI{}).
 		Owns(&manilav1beta1.ManilaScheduler{}).
 		Owns(&manilav1beta1.ManilaShare{}).
+		Owns(&rabbitmqv1.TransportURL{}).
 		Owns(&batchv1.Job{}).
 		Owns(&corev1.ConfigMap{}).
-		Owns(&corev1.Secret{}).
+		// Watch for TransportURL Secrets which belong to any TransportURLs created by Cinder CRs
```
Comment (gouthampacha) [inline controllers/manila_controller.go:243]: Manila CRs

File: controllers/manila_controller.go [hunk 3]
```diff
@@ -94,6 +98,8 @@ type ManilaReconciler struct {
 // +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;create;update;patch;delete;watch
 // +kubebuilder:rbac:groups=mariadb.openstack.org,resources=mariadbdatabases,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=keystone.openstack.org,resources=keystoneapis,verbs=get;list;watch
+// +kubebuilder:rbac:groups=keystone.openstack.org,resources=keystoneapis,verbs=get;list;watch
```
Comment (gouthampacha) [inline controllers/manila_controller.go:101]: this line is repeated

**Review by gouthampacha — APPROVED** [review]
/lgtm 

Thank you for taking me through this code @fmount ; looks great. minor nits inline that we can address in a follow up change. Let's ship it :)

File: controllers/manila_controller.go (hunk 2 above)
Comment (fmount) [reply controllers/manila_controller.go:243]: Done

File: controllers/manila_controller.go (hunk 3 above)
Comment (fmount) [reply controllers/manila_controller.go:101]: good catch thanks!

**Pull request merged**